linode_api4/objects/linode.py

```diff
@@ -764,20 +784,48 @@ def kvmify(self):
 
         return True
 
-    def mutate(self):
+    def mutate(self, allow_auto_disk_resize=True):
         """
         Upgrades this Instance to the latest generation type
         """
-        self._client.post('{}/mutate'.format(Instance.api_endpoint), model=self)
+
+        params = {
+            "allow_auto_disk_resize": allow_auto_disk_resize
+        }
+
+        self._client.post('{}/mutate'.format(Instance.api_endpoint), model=self, data=params)
 
         return True
 
-    def initiate_migration(self):
+    def initiate_migration(self, region=None, upgrade=None):
         """
         Initiates a pending migration that is already scheduled for this Linode
         Instance
         """
-        self._client.post('{}/migrate'.format(Instance.api_endpoint), model=self)
+        params = {
+            "region": region.id if issubclass(type(region), Base) else region,
+            "upgrade": upgrade
+        }
+
+        self._client.post('{}/migrate'.format(Instance.api_endpoint), model=self, data=params)
+
+    def firewalls(self):
+        """
+        View Firewall information for Firewalls associated with this Linode.
+        """
+        self._client.get('{}/firewalls'.format(Instance.api_endpoint), model=self)
+
+    def nodebalancers(self):
+        """
+        View a list of NodeBalancers that are assigned to this Linode and readable by the requesting User.
+        """
+        self._client.get('{}/nodebalancers'.format(Instance.api_endpoint), model=self)
+
+    def volumes(self):
```

Comment:
For these three methods, could we loop through the returned resources, aggregate a list of resource objects, and return it? For example, the `firewalls` function would return a list of `Firewall` objects for each returned firewall.